### PR TITLE
Eliminate Pulp from ros_buildfarm

### PR DESCRIPTION
**Pull request by cottsay**
The createrepo-agent replacement for how we were previously using Pulp is proving to be at least as stable as our Pulp-based solution. It's time to remove all of the unused Pulp-related code from ros_buildfarm.

This absolutely has potential to break things, so I'm going to do some testing on the staging farm prior to marking this PR as ready for review.

### CODE REVIEW

File: ros_buildfarm/release_job.py [hunk 1]
```diff
@@ -139,11 +139,14 @@ def configure_release_jobs(
...
-    if not jenkins:
-        all_job_configs[job_name] = job_config
+    for os_name, _ in platforms:
+        if package_format_mapping[os_name] not in ('rpm',):
+            job_name, job_config = configure_import_package_job(
+                config_url, rosdistro_name, release_build_name,
+                config=config, build_file=build_file, jenkins=jenkins, dry_run=dry_run)
+            if not jenkins:
+                all_job_configs[job_name] = job_config
+            break
```
Comment (nuclearsandwich) [inline ros_buildfarm/release_job.py:149]: The `break` jumps out at me here. I don't foresee us doing so, but would this prevent a release configuration which mixes rpm and deb packages from properly generating import package jobs depending on the order of iteration?

**Review by nuclearsandwich — COMMENTED** [review]
I'm not sure whether this is looking for affirmative 👍🏼 right now or just an early read / second set of eyes.

Are all of the operations under `scripts/release/rpm/` now served directly via createrepo-agent operations?

Overall looks good. The very basic `git grep pulp` test comes back successfully absent. Likewise the upload_destination_credential does not appear to be used anywhere else so removing it seems entirely acceptable.

File: ros_buildfarm/release_job.py (hunk 1 above)
Comment (cottsay) [reply ros_buildfarm/release_job.py:149]: The `break` happens when a non-RPM platform is specified, and RPM platforms are ignored. The order shouldn't matter here - as long as at least 1 specified platform is non-RPM, you'll get an import job.

**Review by nuclearsandwich — APPROVED** [review]
I did a re-review since it had been a while but the previous 👍🏼 still stands. Thanks for getting this updated and ready to merge!

**Pull request merged**